src/planner/insert_plan.cpp

```diff
@@ -147,7 +147,7 @@ InsertPlan::InsertPlan(
         if (query_columns_cnt < table_columns_cnt) {
           for (size_t col_cntr = 0; col_cntr < table_columns_cnt; col_cntr++) {
             auto col = table_columns[col_cntr];
-            if (std::find(query_columns->begin(), query_columns->end(), col.GetName())
+            if (std::find_if(query_columns->begin(), query_columns->end(), [&col](std::unique_ptr<char[]>& x){return col.GetName() == x.get();})
```

Comment:
This is really gross.:

`[&col](std::unique_ptr<char[]>& x){return col.GetName() == x.get();`

The casting seems wrong and overly complicated. Spread it out over multiple lines.